wordpress-parser/xml-to-svelte.js

```diff
@@ -22,34 +22,58 @@ async function parseAndWriteOutput({ inputGlob, outputDir, idToNameOutputFile, d
 	))
 
 	await Promise.all(fileContents.map(xmlString =>
-		parseXmlAndOutputSvelteComponents({ xmlString, outputDir, idToNameOutputFile, downloadImages, imageOutputDir })
+		parseXmlAndOutputSvelteComponents({ xmlString, outputDir, idToNameOutputFile, navigationOutputFile, downloadImages, imageOutputDir })
 	))
 }
 
-async function parseXmlAndOutputSvelteComponents({ xmlString, outputDir, idToNameOutputFile, downloadImages, imageOutputDir }) {
+async function parseXmlAndOutputSvelteComponents({ xmlString, outputDir, idToNameOutputFile, navigationOutputFile, downloadImages, imageOutputDir }) {
 	const doc = parseXml(xmlString)
 	const rss = findFirstChild(doc, `rss`)
 	const channel = findFirstChild(rss, `channel`)
 	const items = channel.children.filter(node => node.name === `item`)
 
+	const navigation = []
+	const categories = {}
+
 	const pageDetails = items.map(itemNode => {
 		const title = extractText(findFirstChild(itemNode, `title`))
 		const id = extractText(findFirstChild(itemNode, `wp:post_name`))
 		const content = extractText(findFirstChild(itemNode, `content:encoded`))
 
+		const category = findFirstChild(itemNode, `category`).attributes.nicename
+		const categoryTitle = extractText(findFirstChild(itemNode, `category`))
+
+		if(categories[category] === undefined) {
+			categories[category] = navigation.length
+			navigation.push({
```

Comment:
Saving the data in this new form isn't the worst thing, but it does increase the bundle size and make this translation script a bit more complicated.

Reply:
I think if you change it to only spit out the ids instead of the names, like this:

```json

	{
		"id": "getting-started",
		"children": [
			"who-should-use-it",
			"how-to-use-it",
			"before-you-start",
			"top-10-terms-to-know"
		]
	},
	{
		"id": "all-about-respirators",
		"children": [
			"etc"
		]
	}
]
```

then you get the data that you need without duplicating anything in separate files (since all of the names can be derived from the ids via `id-to-name.json`.

Reply:
Good idea.  I started out coding the new navigation structure thinking it would completely replace id-to-names, but then I discovered some more code that used id-to-names that I wasn't comfortable refactoring, so I decided to let them live in harmony.